src/crystal/system/unix/time.cr

```diff
@@ -61,25 +66,71 @@ module Crystal::System::Time
     "/usr/share/lib/zoneinfo/",
     "/usr/lib/locale/TZ/",
   }
-  LOCALTIME = "/etc/localtime"
+
+  # Android Bionic C-specific locations. These are files rather than directories
+  # and use a different format (see `Time::Location.read_android_tzdata`).
+  ANDROID_TZDATA_SOURCES = {
+    "/apex/com.android.tzdata/etc/tz/tzdata",
+    "/system/usr/share/zoneinfo/tzdata",
+  }
 
   def self.zone_sources : Enumerable(String)
     ZONE_SOURCES
   end
 
+  def self.android_tzdata_sources : Enumerable(String)
+    ANDROID_TZDATA_SOURCES
+  end
+
   def self.load_iana_zone(iana_name : String) : ::Time::Location?
     nil
   end
 
-  def self.load_localtime : ::Time::Location?
-    if ::File.file?(LOCALTIME) && ::File.readable?(LOCALTIME)
-      ::File.open(LOCALTIME) do |file|
-        ::Time::Location.read_zoneinfo("Local", file)
-      rescue ::Time::Location::InvalidTZDataError
-        nil
+  {% if flag?(:android) %}
+    def self.load_localtime : ::Time::Location?
+      # NOTE: although reading a system property is expensive, we don't cache it
+      # here since it is expected that most code should only ever be calling
+      # `Time::Location.load`, which is already a cached class property, rather
+      # than `.load_local`. Bionic itself caches the property like this:
+      # https://android.googlesource.com/platform/bionic/+/master/libc/private/CachedProperty.h
+      if timezone = getprop("persist.sys.timezone")
+        if path = ::Time::Location.find_android_tzdata_file(android_tzdata_sources)
+          ::File.open(path) do |file|
+            ::Time::Location.read_android_tzdata(file, true) do |name, location|
+              return location if name == timezone
+            end
+          end
+        end
```

Comment:
How about using early returns instead of nesting?

```suggestion
      timezone = getprop("persist.sys.timezone") || return
      path = ::Time::Location.find_android_tzdata_file(android_tzdata_sources) || return
      ::File.open(path) do |file|
        ::Time::Location.read_android_tzdata(file, true) do |name, location|
          return location if name == timezone
        end
      end
```

Reply:
Early returns make code easier to read indeed, although I'd avoid using `|| return` construct, since it visually shifts this pretty important piece of code out of the "focus zone" (left side for the ltr readers) making it less obvious.